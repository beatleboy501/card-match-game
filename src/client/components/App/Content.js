import React from 'react';
import PropTypes from 'prop-types';
import Paper from '@material-ui/core/Paper';
import MemoryGame from '../MemoryGame/index';
import LevelSelector from './LevelSelector';
import styles from '../../styles/MiddleContent.scss';

const propTypes = {
  currentCards: PropTypes.arrayOf(PropTypes.object),
  levels: PropTypes.arrayOf(PropTypes.object),
  fetchCards: PropTypes.func.isRequired,
  onGameStart: PropTypes.func.isRequired,
  onGameComplete: PropTypes.func.isRequired,
  parentRef: PropTypes.func.isRequired,
  cardMatchCount: PropTypes.number.isRequired,
};

const defaultProps = {
  currentCards: null,
  levels: [],
};

const Content = (props) => {
  const {
    currentCards, levels, fetchCards, onGameStart,
    onGameComplete, parentRef, cardMatchCount,
  } = props;
  return (
    <div className={styles.mainPanel}>
      <Paper className={styles.paper} elevation={10}>
        {currentCards ? (
          <MemoryGame
            ref={el => parentRef(el)}
            cardMatchCount={cardMatchCount}
            cards={currentCards}
            onGameStart={onGameStart}
            onGameComplete={onGameComplete}
          />
        ) : (
          <LevelSelector levels={levels} fetchCards={fetchCards} />
        )}
      </Paper>
    </div>
  );
};


Content.propTypes = propTypes;
Content.defaultProps = defaultProps;

export default Content;
