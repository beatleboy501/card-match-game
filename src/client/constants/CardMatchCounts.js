export default Object.freeze({
  THREE_CARD_MATCH: 3,
  TWO_CARD_MATCH: 2,
});
