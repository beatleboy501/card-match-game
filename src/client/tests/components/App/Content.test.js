import React from 'react';
import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Content from '../../../components/App/Content';
import { easyCards, levels } from '../../testUtils';
import CardMatchCounts from '../../../constants/CardMatchCounts';

Enzyme.configure({ adapter: new Adapter() });

const defaultProps = {
  currentCards: easyCards,
  cardMatchCount: CardMatchCounts.TWO_CARD_MATCH,
  fetchCards: jest.fn(() => {}),
  onGameStart: jest.fn(() => {}),
  onGameComplete: jest.fn(() => {}),
  parentRef: jest.fn(() => {}),
  levels,
};
const mockStore = configureStore();
let component;

it('Renders without crashing', () => {
  component = shallow(<Content {...defaultProps} />).shallow();
  expect(component).toMatchSnapshot();
});

it('Displays level selector when no current cards', () => {
  const props = defaultProps;
  props.currentCards = null;
  component = mount(
    <Provider store={mockStore(props)}>
      <Content {...props} />
    </Provider>,
  );
  expect(component.html()).not.toContain('div class="Flipper"');
  expect(component.html()).toContain('div class="level-selector"');
});

it('Displays game when current cards are set', () => {
  const props = defaultProps;
  props.currentCards = easyCards;
  component = mount(
    <Provider store={mockStore(props)}>
      <Content {...props} />
    </Provider>,
  );
  expect(component.html()).toContain('div class="Flipper"');
  expect(component.html()).not.toContain('div class="level-selector"');
});
