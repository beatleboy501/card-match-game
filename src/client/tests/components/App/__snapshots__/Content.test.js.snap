// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <div
    className="mainPanel"
  >
    <WithStyles(Paper)
      className="paper"
      elevation={10}
    >
      <Connect(MemoryGame)
        cardMatchCount={2}
        cards={
          Array [
            Object {
              "id": 1,
              "value": "a",
            },
            Object {
              "id": 2,
              "value": "b",
            },
            Object {
              "id": 3,
              "value": "c",
            },
            Object {
              "id": 4,
              "value": "d",
            },
            Object {
              "id": 5,
              "value": "a",
            },
            Object {
              "id": 6,
              "value": "b",
            },
            Object {
              "id": 7,
              "value": "c",
            },
            Object {
              "id": 8,
              "value": "d",
            },
          ]
        }
        onGameComplete={[MockFunction]}
        onGameStart={[MockFunction]}
      />
    </WithStyles(Paper)>
  </div>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <WithStyles(Paper)
        className="paper"
        elevation={10}
      >
        <Connect(MemoryGame)
          cardMatchCount={2}
          cards={
            Array [
              Object {
                "id": 1,
                "value": "a",
              },
              Object {
                "id": 2,
                "value": "b",
              },
              Object {
                "id": 3,
                "value": "c",
              },
              Object {
                "id": 4,
                "value": "d",
              },
              Object {
                "id": 5,
                "value": "a",
              },
              Object {
                "id": 6,
                "value": "b",
              },
              Object {
                "id": 7,
                "value": "c",
              },
              Object {
                "id": 8,
                "value": "d",
              },
            ]
          }
          onGameComplete={[MockFunction]}
          onGameStart={[MockFunction]}
        />
      </WithStyles(Paper)>,
      "className": "mainPanel",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Connect(MemoryGame)
          cardMatchCount={2}
          cards={
            Array [
              Object {
                "id": 1,
                "value": "a",
              },
              Object {
                "id": 2,
                "value": "b",
              },
              Object {
                "id": 3,
                "value": "c",
              },
              Object {
                "id": 4,
                "value": "d",
              },
              Object {
                "id": 5,
                "value": "a",
              },
              Object {
                "id": 6,
                "value": "b",
              },
              Object {
                "id": 7,
                "value": "c",
              },
              Object {
                "id": 8,
                "value": "d",
              },
            ]
          }
          onGameComplete={[MockFunction]}
          onGameStart={[MockFunction]}
        />,
        "className": "paper",
        "elevation": 10,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "cardMatchCount": 2,
          "cards": Array [
            Object {
              "id": 1,
              "value": "a",
            },
            Object {
              "id": 2,
              "value": "b",
            },
            Object {
              "id": 3,
              "value": "c",
            },
            Object {
              "id": 4,
              "value": "d",
            },
            Object {
              "id": 5,
              "value": "a",
            },
            Object {
              "id": 6,
              "value": "b",
            },
            Object {
              "id": 7,
              "value": "c",
            },
            Object {
              "id": 8,
              "value": "d",
            },
          ],
          "onGameComplete": [MockFunction],
          "onGameStart": [MockFunction],
        },
        "ref": [Function],
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <WithStyles(Paper)
          className="paper"
          elevation={10}
        >
          <Connect(MemoryGame)
            cardMatchCount={2}
            cards={
              Array [
                Object {
                  "id": 1,
                  "value": "a",
                },
                Object {
                  "id": 2,
                  "value": "b",
                },
                Object {
                  "id": 3,
                  "value": "c",
                },
                Object {
                  "id": 4,
                  "value": "d",
                },
                Object {
                  "id": 5,
                  "value": "a",
                },
                Object {
                  "id": 6,
                  "value": "b",
                },
                Object {
                  "id": 7,
                  "value": "c",
                },
                Object {
                  "id": 8,
                  "value": "d",
                },
              ]
            }
            onGameComplete={[MockFunction]}
            onGameStart={[MockFunction]}
          />
        </WithStyles(Paper)>,
        "className": "mainPanel",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Connect(MemoryGame)
            cardMatchCount={2}
            cards={
              Array [
                Object {
                  "id": 1,
                  "value": "a",
                },
                Object {
                  "id": 2,
                  "value": "b",
                },
                Object {
                  "id": 3,
                  "value": "c",
                },
                Object {
                  "id": 4,
                  "value": "d",
                },
                Object {
                  "id": 5,
                  "value": "a",
                },
                Object {
                  "id": 6,
                  "value": "b",
                },
                Object {
                  "id": 7,
                  "value": "c",
                },
                Object {
                  "id": 8,
                  "value": "d",
                },
              ]
            }
            onGameComplete={[MockFunction]}
            onGameStart={[MockFunction]}
          />,
          "className": "paper",
          "elevation": 10,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "cardMatchCount": 2,
            "cards": Array [
              Object {
                "id": 1,
                "value": "a",
              },
              Object {
                "id": 2,
                "value": "b",
              },
              Object {
                "id": 3,
                "value": "c",
              },
              Object {
                "id": 4,
                "value": "d",
              },
              Object {
                "id": 5,
                "value": "a",
              },
              Object {
                "id": 6,
                "value": "b",
              },
              Object {
                "id": 7,
                "value": "c",
              },
              Object {
                "id": 8,
                "value": "d",
              },
            ],
            "onGameComplete": [MockFunction],
            "onGameStart": [MockFunction],
          },
          "ref": [Function],
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
